ui/src/message_popup/experience_page.jsx

```diff
@@ -117,24 +98,32 @@ export default React.createClass({
     this.setState(this.getInitialState());
   },
 
+  onSaveScaffoldingAndSession(scaffolding, gameSession){
```

Comment:
I think this second parameter isn't quite a full gameSession (ie., it doesn't have the `responseTimes` and the `sessionId` yet.  This method could take the scaffolding and then the email and set of questions, and it's job is to create a `gameSession` from those pieces and the other pieces needed.